pkg/controller/mysqlcluster/internal/syncer/config_map.go

```diff
@@ -53,14 +54,47 @@ func NewConfigMapSyncer(c client.Client, scheme *runtime.Scheme, cluster *mysqlc
 			return fmt.Errorf("failed to create mysql configs: %s", err)
 		}
 
+		preStopSh := buildBashPreStop()
+
 		cm.Data = map[string]string{
-			"my.cnf": data,
+			"my.cnf":      data,
+			shPreStopFile: preStopSh,
 		}
 
 		return nil
 	})
 }
 
+func buildBashPreStop() string {
+	data := `#!/bin/bash
+set -ex
+
+current=$(date "+%Y-%m-%d %H:%M:%S")
+echo "[${current}]preStop is ongoing"
+read_only_status=$(mysql --defaults-file=ConfClientPathHolder -NB -e 'SELECT @@read_only')
+replica_status=$(mysql --defaults-file=ConfClientPathHolder -NB -e 'show slave status\G')
+# orchestrator will isolate old master during failover
+has_replica_hosts=$(mysql --defaults-file=ConfClientPathHolder -NB -e 'show slave hosts\G')
+replica_status_count=$(echo -n "$replica_status" | wc -l )
+has_replica_count=$(echo -n "$has_replica_hosts" | wc -l )
+echo "hostname=$(hostname) readonly=${read_only_status} show_slave_status=${replica_status_count}"
+echo "has_replica_hosts=${has_replica_count}"
+if [ ${read_only_status} -eq 0  ] && [ ${replica_status_count} -eq 0 ] && [ ${has_replica_count} -gt 0 ]
+then
+		masterhostname=$( curl  -s "${ORCH_HTTP_API}/master/${ORCH_CLUSTER_ALIAS}" |  awk -F":" '{print $3}' | awk -F'"' '{print $2}' )
+        echo "master from orchestrator: ${masterhostname}"
+        if [ "${FQDN}" == "${masterhostname}" ]
+        then
+                curl  -s "${ORCH_HTTP_API}/graceful-master-takeover-auto/${ORCH_CLUSTER_ALIAS}"
```

Comment:
It might block, if orchestrator or the network don't work well, but terminationGracePeriodSeconds of pod terminating phase is 30 seconds, I think it's ok . If you wanna set the timeout of curl, 
timeout 10 curl -s ""